app/assets/javascripts/discourse/app/services/search.js

```diff
@@ -20,6 +21,27 @@ export default class Search extends Service {
   // only relative for the widget search menu
   searchContextEnabled = false; // checkbox to scope search
 
+  get currentSearchInputId() {
+    if (this.welcomeBannerSearchInViewport) {
+      return "welcome-banner-search-input";
+    } else if (this.searchExperience === "search_field") {
+      return "header-search-input";
+    } else {
+      return "icon-search-input";
+    }
+  }
+
+  get searchExperience() {
+    return applyValueTransformer(
+      "site-setting-search-experience",
+      this.siteSettings.search_experience
+    );
+  }
+
+  focusSearchInput() {
+    document.getElementById(this.currentSearchInputId).focus();
```

Comment:
How stable is the assumption that one of the element IDs can be found? Would it be worth doing a safe navigation call to `#focus`?

```suggestion
    document.getElementById(this.currentSearchInputId)?.focus();
```

Reply:
Yep fair